omnipaxos/src/ballot_leader_election.rs

```diff
@@ -114,21 +112,20 @@ impl BallotLeaderElection {
         let peers = config.peers;
         let num_nodes = &peers.len() + 1;
         let quorum = Quorum::with(config.flexible_quorum, num_nodes);
-        let mut initial_ballot = Ballot::with(config_id, 1, config.priority, pid);
+        let initial_ballot = match recovered_leader {
+            Some(b) if b == Ballot::default() => {
+                Ballot::with(config_id, INITIAL_ROUND, config.priority, pid)
+            }
+            // Prevents a recovered server from retain BLE leadership with the same ballot.
+            Some(_) => Ballot::with(config_id, RECOVERY_ROUND, config.priority, pid),
+            None => Ballot::with(config_id, INITIAL_ROUND, config.priority, pid),
+        };
         let initial_leader = match recovered_leader {
-            Some(b) if b == Ballot::default() => initial_ballot,
+            Some(b) if b == Ballot::default() => {
+                Ballot::with(config_id, INITIAL_ROUND, config.priority, pid)
+            }
             Some(b) => b,
-            None => initial_ballot,
```

Comment:
instead of pattern matching twice, do it once and return a tuple `let (initial_ballot, initial_leader) = match { ... }`